lib/widgets/lightbox.dart

```diff
@@ -89,6 +94,116 @@ class _CopyLinkButton extends StatelessWidget {
   }
 }
 
+class _DownloadImageButton extends StatelessWidget {
+  const _DownloadImageButton({required this.url});
+  final Uri url;
+
+  Future<bool> _saveImageToGeneralStorage(BuildContext context, Uri imageUrl) async {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    // Retrieve account info and construct headers for authorization
+    final store = PerAccountStoreWidget.of(context);
+    final headers = {
+      if (imageUrl.origin == store.account.realmUrl.origin)
+        ...authHeader(email: store.account.email, apiKey: store.account.apiKey),
+      ...userAgentHeader(),
+    };
+
+    // Request storage permission
+    final externalStoragePermission = await Permission.manageExternalStorage.status;
+    if (!externalStoragePermission.isGranted) {
+      final externalStatus = await Permission.manageExternalStorage.request();
+      if (externalStatus.isPermanentlyDenied) {
+        if (!context.mounted) return false;
+
+        final bool openSettings = await showDialog(
+          context: context,
+          builder: (context) => AlertDialog(
+            title: Text(zulipLocalizations.permissionDeniedTitle),
+            content: Text(zulipLocalizations.storagePermissionPermanentlyDenied),
+            actions: [
+              TextButton(
+                onPressed: () => Navigator.pop(context, false),
+                child: Text(zulipLocalizations.cancel),
+              ),
+              TextButton(
+                onPressed: () => Navigator.pop(context, true),
+                child: Text(zulipLocalizations.openSettings),
+              ),
+            ],
+          ),
+        ) ?? false;
+
+        if (openSettings) {
+          await openAppSettings();
+        }
+        return false;
+      }
+
+      if (!externalStatus.isGranted) {
+        if (!context.mounted) return false;
+
+        ScaffoldMessenger.of(context).showSnackBar(
+          SnackBar(content: Text(zulipLocalizations.storagePermissionRequired)),
+        );
+        return false;
+      }
+    }
+
+    try {
+      // Download the image with headers and a timeout of 30 seconds
+      final response = await http
+          .get(imageUrl, headers: headers)
+          .timeout(const Duration(seconds: 30));
+
+      if (response.statusCode != 200) {
+        throw Exception(zulipLocalizations.imageDownloadFailed);
+      }
+
+      // Save the downloaded file to the Downloads directory
+      final directory = Directory('/storage/emulated/0/Download');
+      if (!directory.existsSync()) {
+        directory.createSync(recursive: true);
+      }
+
+      final filePath = '${directory.path}/${imageUrl.pathSegments.last}';
```

Comment:
For path manipulation, we should use `path.join`. Search in `database.dart` for an example.